`ignore-from-file` ignored in `rule` definition

```
$ yamllint --version
yamllint 1.29.0
```
### Description

I have a project where a lot of yaml files have specific rules that need to be ignored.

In order to avoid duplicate code, I want to use the `ignore-from-file` feature in the rule definition but it is ignored.


```
$ cat .yamllint
---
extends: default

rules:
  empty-lines:
    ignore-from-file: .yamlignore
```
```
$ cat .yamlignore
/ignore-empty-lines.yml
```
```
$ cat ignore-empty-lines.yml
---


# These empty lines above just exist.
```

### Expected result

Yamllint should ignore this rule on this file.

### Actual result
```
$ yamllint .
./ignore-empty-lines.yml
  4:1       error    too many blank lines (3 > 2)  (empty-lines)
```


### Workaround
Use the `ignore` keyword. I have a lot of files that need to have multiple errors ignored. The final config file looks like this which leads to duplicate code:

```
$ cat .yamllint
---
extends: default

rules:
  empty-lines:
    ignore:
      - /ignore1.yml
      - /ignore2.yml
      - ...
      - /ignore42.yml
  comments:
    ignore:
      - /ignore1.yml
      - /ignore2.yml
      - ...
      - /ignore42.yml
```
___

The [documentation ](https://yamllint.readthedocs.io/en/v1.29.0/configuration.html#ignoring-paths) does not specify that this keyword is ignored in a `rules` block and the example implicitly gives the impression that it should work.

Either the documentation should be explicit about this behavior or yamllint should work as described above.

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -212,6 +212,14 @@ class SimpleConfigTestCase(unittest.TestCase):
                                   '  colons:\n'
                                   '    ignore: yes\n')
 
+    def test_invalid_rule_ignore_from_file(self):
+        self.assertRaises(
+            config.YamlLintConfigError,
+            config.YamlLintConfig,
+            'rules:\n'
+            '  colons:\n'
+            '    ignore-from-file: 1337\n')
+
     def test_invalid_locale(self):
         with self.assertRaisesRegex(
                 config.YamlLintConfigError,
@@ -660,11 +668,18 @@ class IgnoreConfigTestCase(unittest.TestCase):
     def test_run_with_ignore_from_file(self):
         with open(os.path.join(self.wd, '.yamllint'), 'w') as f:
             f.write('extends: default\n'
-                    'ignore-from-file: .gitignore\n')
+                    'ignore-from-file: .gitignore\n'
+                    'rules:\n'
+                    '  key-duplicates:\n'
+                    '    ignore-from-file: .ignore-key-duplicates\n')
+
         with open(os.path.join(self.wd, '.gitignore'), 'w') as f:
             f.write('*.dont-lint-me.yaml\n'
                     '/bin/\n'
                     '!/bin/*.lint-me-anyway.yaml\n')
+
+        with open(os.path.join(self.wd, '.ignore-key-duplicates'), 'w') as f:
+            f.write('/ign-dup\n')
 
         sys.stdout = StringIO()
         with self.assertRaises(SystemExit):
@@ -686,10 +701,8 @@ class IgnoreConfigTestCase(unittest.TestCase):
             './file-at-root.yaml:3:3: ' + keydup,
             './file-at-root.yaml:4:17: ' + trailing,
             './file-at-root.yaml:5:5: ' + hyphen,
-            './ign-dup/file.yaml:3:3: ' + keydup,
             './ign-dup/file.yaml:4:17: ' + trailing,
             './ign-dup/file.yaml:5:5: ' + hyphen,
-            './ign-dup/sub/dir/file.yaml:3:3: ' + keydup,
             './ign-dup/sub/dir/file.yaml:4:17: ' + trailing,
             './ign-dup/sub/dir/file.yaml:5:5: ' + hyphen,
             './ign-trail/file.yaml:3:3: ' + keydup,

--- a/yamllint/config.py
+++ b/yamllint/config.py
@@ -153,8 +153,21 @@ def validate_rule_conf(rule, conf):
         return False
 
     if isinstance(conf, dict):
-        if ('ignore' in conf and
-                not isinstance(conf['ignore'], pathspec.pathspec.PathSpec)):
+        if ('ignore-from-file' in conf and not isinstance(
+                conf['ignore-from-file'], pathspec.pathspec.PathSpec)):
+            if isinstance(conf['ignore-from-file'], str):
+                conf['ignore-from-file'] = [conf['ignore-from-file']]
+            if not (isinstance(conf['ignore-from-file'], list)
+                    and all(isinstance(line, str)
+                            for line in conf['ignore-from-file'])):
+                raise YamlLintConfigError(
+                    'invalid config: ignore-from-file should contain '
+                    'valid filename(s), either as a list or string')
+            with fileinput.input(conf['ignore-from-file']) as f:
+                conf['ignore'] = pathspec.PathSpec.from_lines(
+                    'gitwildmatch', f)
+        elif ('ignore' in conf and not isinstance(
+                conf['ignore'], pathspec.pathspec.PathSpec)):
             if isinstance(conf['ignore'], str):
                 conf['ignore'] = pathspec.PathSpec.from_lines(
                     'gitwildmatch', conf['ignore'].splitlines())

